ExpressLRS/ExpressLRS
# Always use flashed options after a new flash

When flashing a device, it should always use the options supplied in configurator.

When flashing we now generate a random string which is written to the option JSON object. If this is different to the one on the SPIFFS partition then we use the flashed one. This means a user can modify their settings in the web UI and the new discriminator string is written to the SPIFFS and will use the SPIFSS options until the device is flashed again.

Check the "device-id" when saving options and if it is different it will return an error telling the user they need to refresh.
Disables the "Save & Reboot" button until a config has been read from the device.

Also fixes airport settings not being done when using the binary_configurator (for cloudbuilds).

This required an upgrade to the base ESP8266 platform as there was a bug when serialising JSON to the SPIFFS filesystem! It would write the correct number of bytes, but the last few were 0xFF ?! The code change was required because we now check the device-id on options save.

Fixes #2121 and #2120

## Test 1
1. Flash a TX or RX with or without a passphrase
2. Go to the web-UI and change the passphrase
3. Reboot the TX or RX and check in the web-UI that the new UID for the passphrase is shown
4. Flash the TX or RX again, with the original passphrase
5. Verify that the web-UI shows the UID for the original passphrase

## Test 2
1. Flash 2 TX or RX modules without home network
2. Start device 1, and connect to the web UI
3. Leave the web-page open and power off device 1 and power on device 2
4. Connect to the AP of device 2 making sure not to refresh the browser
5. Try to save settings and you should get an error about a mismatched device and to refresh the browser
